Cython/Utility/ModuleSetupCode.c

```diff
@@ -194,6 +194,9 @@
   #ifdef Py_LIMITED_API
     #undef __PYX_LIMITED_VERSION_HEX
     #define __PYX_LIMITED_VERSION_HEX Py_LIMITED_API
+  #elif CYTHON_LIMITED_API >= 0x03000000
```

Comment:
@scoder This is IMO the only controversial part of this PR. What do you think?

Reply:
If we're doing this I think I'd change it to:

```
#if CYTHON_LIMITED_API != 1
    #undef __PYX_LIMITED_VERSION_HEX
    #define __PYX_LIMITED_VERSION_HEX CYTHON_LIMITED_API
#elif defined(Py_LIMITED_API)
    #undef __PYX_LIMITED_VERSION_HEX
    #define __PYX_LIMITED_VERSION_HEX Py_LIMITED_API
#endif
```

That makes user-preference override `Py_LIMITED_API` in general. And treats `1` as the only special value.

But wait to see what @scoder thinks...

Reply:
In principle, I agree with you. However, what if both `CYTHON_LIMITED_API` and `Py_LIMITED_API` are defined to different values? The case `CYTHON_LIMITED_API <= Py_LIMITED_API` is probably OK, but I may be missing something. Should will add a check for that and `#error` if things are off?

Reply:
My thought is that `CYTHON_LIMITED_API <= Py_LIMITED_API` should work and might be useful to some people (maybe just for testing?). `CYTHON_LIMITED_API > Py_LIMITED_API` probably won't work but if users' want to request it then that's their choice.

I really don't know how useful this configurability is likely to be though.

Reply:
Under normal circumstances, it should be enough to define `Py_LIMITED_API=0xV3R510N`, so that should be easy and "just work". While I also consider `CYTHON_LIMITED_API` more specific, I don't see why we should allow users to let it contradict the general CPython macro. It just leads to unrealistic build scenarios, even if we consider them test setups.

I'm fine with keeping `CYTHON_LIMITED_API` as a flag and requiring `Py_LIMITED_API` to set a specific target version.